misc/hooks/pre-commit-clang-format

```diff
@@ -99,11 +99,17 @@ if [ ! -x "$CLANG_FORMAT" ] ; then
     exit 1
 fi
 
-CLANG_FORMAT_VERSION="$(clang-format --version | cut -d' ' -f3)"
-CLANG_FORMAT_MAJOR="$(echo "$CLANG_FORMAT_VERSION" | cut -d'.' -f1)"
+# Example output strings reported by `clang-format --version`:
+# Ubuntu: Ubuntu clang-format version 11.0.0-2
+# Fedora: clang-format version 11.0.0 (Fedora 11.0.0-2.fc33)
+#
+# Due to inconsistencies, we have to be careful:
+# - Get everything until the first period.
+# - Reverse, get the first word, reverse it again (this effectively gets the *last* word).
+CLANG_FORMAT_MAJOR="$(clang-format --version | cut -d. -f1 | rev | cut -d" " -f1 | rev)"
```

Comment:
Might be more reliable to use sed, grep or perl instead of relying on a specific order of words.
```
$ clang-format --version | sed "s/[^0-9\.]*\([0-9\.]*\).*/\1/"
12.0.0
```

Reply:
That `sed` command works correctly on the Ubuntu version string too, so I guess I'll use that instead.